src/server.ts

```diff
@@ -97,6 +98,13 @@ export function getServer(
   // http://expressjs.com/en/advanced/best-practice-security.html#at-a-minimum-disable-x-powered-by-header
   app.disable('x-powered-by');
 
+  // If a Pub/Sub subscription is configured to invoke a user's function directly, the request body
+  // needs to be marshalled into the structure that wrapEventFunction expects. This unblocks local
+  // development with the Pub/Sub emulator
+  if (functionSignatureType === SignatureType.EVENT) {
```

Comment:
Don't we need to add `PUBSUB_EMULATOR_HOST` check somewhere?

Reply:
You are right, my original idea was that this would check `process.env['PUBSUB_EMULATOR_HOST'] != null` here, but I wanted to put this up as an idea after thinking about @di's comments offline. Here we just treat this as any event conversion, where we detect the payload structure does not match the requested signature type and marshal it. This removes friction from the dev workflow: it just works out of the box without additional setup. It also improves portability of event functions (I can point my Pub/Sub push subscription at a function running in on any server with this change). The cost is some risk that we marshal an even that we aren't supposed to, but I think the benefits outweigh the risk as the current behaviour is clearly broken.

What do you think?

Reply:
(This makes sense to me!)

Reply:
Hmm.

Isn't the supervisor modifying GCF payloads such that GCF Pub/Sub prod works fine but FF Pub/Sub local doesn't? As long as we're sure that this version of the FF will work with prod, I suppose we're fine.

Perhaps `isRawPubSubRequestBody` evals to false in GCF.

Reply:
> Perhaps `isRawPubSubRequestBody` evals to false in GCF.

Yes exactly, if the supervisor has already marshalled the payload, the request body will be different, and this middleware will be a no-op.